src/tests/unit/tests/injected/all-frame-runner.test.ts

```diff
@@ -0,0 +1,314 @@
+            topWindowCallbackMock.setup(m => m(fakePayload)).verifiable(Times.once());
+
+            windowUtilsMock
+                .setup(m => m.isTopWindow())
+                .returns(() => true)
+                .verifiable(Times.once());
+
+            await providedCallback(fakePayload);
+
+            topWindowCallbackMock.verifyAll();
+        });
+    });
+
+    test('start: calls start in current frame & sends start command to other frames', async () => {
+        const frameRunner = getFrameRunnerInstance(
+            frameMessengerMock,
+            htmlElementUtilsMock,
+            windowUtilsMock,
+            unitTestListenerMock,
+        );
+
+        unitTestListenerMock.setup(m => m.start()).verifiable(Times.once());
+
+        setupSendCommandToFrames(htmlElementUtilsMock, frameMessengerMock, {
+            command: (frameRunner as any).startCommand,
+        });
+
+        await frameRunner.start();
+    });
+
+    test('stop: calls stop in current frame & sends stop command to other frames', async () => {
+        const frameRunner = getFrameRunnerInstance(
+            frameMessengerMock,
+            htmlElementUtilsMock,
+            windowUtilsMock,
+            unitTestListenerMock,
+        );
+
+        unitTestListenerMock.setup(m => m.stop()).verifiable(Times.once());
+
+        setupSendCommandToFrames(htmlElementUtilsMock, frameMessengerMock, {
+            command: (frameRunner as any).stopCommand,
+        });
+
+        await frameRunner.stop();
+    });
+
+    const setupSendCommandToFrames = (
+        htmlUtilsMock: IMock<HTMLElementUtils>,
+        frameMsgrMock: IMock<FrameMessenger>,
+        expectedCommandMessage: CommandMessage,
+    ) => {
+        const fakeFrames = [{ id: 'iframe1' }, { id: 'iframe2' }];
+        htmlUtilsMock
+            .setup(m => m.getAllElementsByTagName('iframe'))
+            .returns(() => fakeFrames as any)
+            .verifiable(Times.once());
+
+        fakeFrames.forEach(f => {
+            frameMsgrMock
+                .setup(m =>
+                    m.sendMessageToFrame(f as unknown as HTMLIFrameElement, expectedCommandMessage),
+                )
+                .verifiable(Times.once());
+        });
+    };
+
+    describe('on child result: OnResultFromChildFrame', () => {
+        const fakeFrames = [{ id: 'iframe1' }, { id: 'iframe2' }] as HTMLIFrameElement[];
+        const fakeWindow = {} as Window;
+        const fakeCm: CommandMessage = {
+            command: 'fake-command',
+            payload: {
+                id: 'fake-payload',
+            },
+        };
+        test('calls topWindowCallback if in main window', async () => {
+            const topWindowCallbackMock = Mock.ofInstance((_: UnitTestTargetType) => {});
+
+            const { commandFunc } = captureOnResultFromChildFrameCallback(
+                unitTestListenerMock,
+                topWindowCallbackMock.object,
+            );
+
+            htmlElementUtilsMock
```

Comment:
Might not be a bad idea to refactor some of these setups to functions to reuse in multiple tests.